### PR TITLE
construct the main performance curve function 

### PR DESCRIPTION
**Pull request by trant22t**

### CODE REVIEW

File: performance_curves/utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from typing import Tuple
+
+import numpy as np
+
+
+def synchronize_sort(key: np.ndarray, dependent: np.ndarray, ascending=False) -> Tuple[np.ndarray, np.ndarray]:
+    """
```
Comment (erp12) [inline performance_curves/utils.py:7]: Same comment re one-liner description.

File: performance_curves/performance_curve.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    Return:
+        sorted_y_score: array of float, shape = [n_samples] or [n_bins]
+            In case num_bins is `None`, return probability estimates of all data points sorted in descending order.
+            In case num_bins is not `None`, return upper bounds of the bins sorted in descending order.
+        results: array of float, shape = [n_samples] or [n_bins]
+            Performance values of metrics of interest when setting the cutoff at the corresponding y_score
+
+    """
+
+    assert len(y_score) == len(y_true)
```
Comment (erp12) [inline performance_curves/performance_curve.py]: We should probably add a custom error message here.

File: performance_curves/performance_curve.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        plt.show()
+
+
+def performance_curve(
+    y_true: np.ndarray,
+    y_score: np.ndarray,
+    metric: Callable[[np.ndarray, np.ndarray], float],
+    num_bins: Optional[int] = None
+) -> PerformanceCurve:
+    """
```
Comment (erp12) [inline performance_curves/performance_curve.py:33]: Docstrings should have a one-liner description.

**Review by erp12 — CHANGES_REQUESTED** [review]
Woops! Looks like I forgot to install sklearn on the CI/CD tests. If you rebase on master the checks should stop failing.

Code looks very clean. I have just have a couple notes on documentation and the assert error message.

File: performance_curves/utils.py (hunk 1 above)
Comment (trant22t) [reply performance_curves/utils.py:7]: Added!

File: performance_curves/performance_curve.py (hunk 2 above)
Comment (trant22t) [reply performance_curves/performance_curve.py]: Added!

File: performance_curves/performance_curve.py (hunk 3 above)
Comment (trant22t) [reply performance_curves/performance_curve.py:33]: Added!

**Review by erp12 — APPROVED** [review]

**Pull request merged**